lib/cookpad/performance/engine.rb

```diff
@@ -7,6 +7,8 @@ class Engine < ::Rails::Engine
         if !Rails.env.production? &&
            (ENV["LOG_N_PLUS_ONE_QUERIES"] == "true" ||
              ENV["RAISE_N_PLUS_ONE_QUERIES"] == "true")
+
+          require Cookpad::Performance::Engine.root.join("app/controllers/concerns/n_plus_one_detection")
           ActionController::Base.include(NPlusOneDetection)
```

Comment:
I think the problem might be because you refer to this concern by its class name and not the full class name with module.

I was looking at https://github.com/thredded/thredded/blob/37faa5eb67889a03ef02d80ffe41aa97a92560fa/lib/thredded/engine.rb#L14-L15 and seeing that they don't need to do a require.

So, I wonder, will this work?

```suggestion
          ActionController::Base.include(Cookpad::Performance::NPlusOneDetection)
```

Reply:
I gave this a go, but it didn't work. The module isn't nested within `Cookpad::Performance`, so that shouldn't affect it.

Also, note, I created a vanilla Rails 7.0 app and gem that has a concern inside. The same problem didn't occur there at all  ¯\_(ツ)_/¯

Reply:
As soon as I wrote that, I realised that `isolate_namespace` is probably part of the issue her. Once I correctly namespaced the concern, the issue was resolved.